data/tutorials/tt_02_file_manipulation.md

```diff
@@ -9,123 +9,618 @@ date: 2021-05-27T21:07:30-00:00
-* channels that read from a file: type `in_channel`
-
-### Writing
-For writing into a file, you would do this:
-
-1. Open the file to obtain an `out_channel`
-1. Write to the channel
-1. If you want to force writing to the physical device, you must flush
- the channel, otherwise writing will not take place immediately.
-1. When you are done, you can close the channel. This flushes the
- channel automatically.
-
-Commonly used functions: `open_out`, `open_out_bin`, `flush`,
-`close_out`, `close_out_noerr`
-
-Standard `out_channel`s: `stdout`, `stderr`
-
-### Reading
-For reading data from a file you would do this:
-
-1. Open the file to obtain an `in_channel`
-1. Read characters from the channel. Reading consumes the channel, so
- if you read a character, the channel will point to the next
- character in the file.
-1. When there are no more characters to read, the `End_of_file`
- exception is raised. Often, this is where you want to close the
- channel.
-
-Commonly used functions: `open_in`, `open_in_bin`, `close_in`,
-`close_in_noerr`
-
-Standard `in_channel`: `stdin`
-
-### Seeking
-Whenever you write or read something to or from a channel, the current
-position changes to the next character after what you just wrote or
-read. Occasionally, you may want to skip to a particular position in the
-file, or restart reading from the beginning. This is possible for
-channels that point to regular files, use `seek_in` or `seek_out`.
-
-### Gotchas
-* Don't forget to flush your `out_channel`s if you want to actually
- write something. This is particularly important if you are writing
- to non-files such as the standard output (`stdout`) or a socket.
-* Don't forget to close any unused channel, because operating systems
- have a limit on the number of files that can be opened
- simultaneously. You must catch any exception that would occur during
- the file manipulation, close the corresponding channel, and re-raise
- the exception.
-* The `Unix` module provides access to non-buffered file descriptors
- among other things. It provides standard file descriptors that have
- the same name as the corresponding standard channels: `stdin`,
- `stdout` and `stderr`. Therefore if you do `open Unix`, you may get
- type errors. If you want to be sure that you are using the `stdout`
- channel and not the `stdout` file descriptor, you can prepend it
- with the module name where it comes from: `Stdlib.stdout`. *Note
- that most things that don't seem to belong to any module actually
- belong to the `Stdlib` module, which is automatically opened.*
-* `open_out` and `open_out_bin` truncate the given file if it already
- exists! Use `open_out_gen` if you want an alternate behavior.
-
-### Example
-
-<!-- $MDX file=examples/file_manip.ml -->
-```ocaml
-let file = "example.dat"
-let message = "Hello!"
+File manipulation includes operations such as creating new files and
+directories, reading and writing from files, deleting files, etc.
```

Comment:
This might be out-of-scope for this PR but it looks like all the commands (and file paths) are specified for Unix systems.

OCaml doesn't have great Windows support, unfortunately, but I think we can be mindful about Windows users and at least mention in the beginning something like "this tutorial focuses on Unix file system operations however many of them should work on Windows as well".

Reply:
I think this is a great point. The need for Windows tutorials will only increase with the release of opam 2.2 and OCaml gaining better Windows support.